src/resolution/common.rs

```diff
@@ -0,0 +1,248 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use deno_semver::Version;
+use deno_semver::VersionReq;
+use once_cell::sync::Lazy;
+use thiserror::Error;
+
+use crate::registry::NpmPackageInfo;
+use crate::registry::NpmPackageVersionInfo;
+
+pub static LATEST_VERSION_REQ: Lazy<VersionReq> =
+  Lazy::new(|| VersionReq::parse_from_specifier("latest").unwrap());
+
+#[derive(Debug, Error, Clone)]
+pub enum NpmPackageVersionResolutionError {
```

Comment:
Added this error type in this PR. Will follow up elsewhere with using `thiserror` more.